apps/ledger-live-mobile/e2e/models/DeviceAction.ts

```diff
@@ -0,0 +1,216 @@
+import BigNumber from "bignumber.js";
+import { DeviceModelId } from "@ledgerhq/types-devices";
+import {
+  deviceInfo155 as deviceInfo,
+  deviceInfo210lo5,
+  mockListAppsResult as innerMockListAppResult,
+} from "@ledgerhq/live-common/apps/mock";
+import { AppOp } from "@ledgerhq/live-common/apps/types";
+import { AppType, DeviceInfo } from "@ledgerhq/types-live/lib/manager";
+import { Device } from "@ledgerhq/live-common/hw/actions/types";
+import { Transaction } from "@ledgerhq/live-common/generated/types";
+import { waitForElementById, tapById, delay } from "../helpers";
+import { mockDeviceEvent } from "../bridge/server";
+import { DeviceLike } from "../../src/reducers/types";
+
+const mockListAppsResult = (
+  appDesc: string,
+  installedDesc: string,
+  deviceInfo: DeviceInfo,
+  deviceModelId?: DeviceModelId,
+) => {
+  const result = innerMockListAppResult(appDesc, installedDesc, deviceInfo, deviceModelId);
+  Object.keys(result?.appByName).forEach(key => {
+    result.appByName[key] = { ...result.appByName[key], type: AppType.currency };
+  });
+  return result;
+};
+
+// this implement mock for the "legacy" device action (the one working with live-common/lib/hw/actions/*)
+export default class DeviceAction {
+  deviceLike: DeviceLike;
+  device: Device;
+
+  constructor(deviceLike: DeviceLike) {
+    this.deviceLike = deviceLike;
+    this.device = this.deviceLikeToDevice(deviceLike);
+  }
+
+  deviceLikeToDevice(d: DeviceLike): Device {
+    return {
+      deviceId: d.id,
+      deviceName: d.name,
+      modelId: d.modelId,
+      wired: false,
+    };
+  }
+
+  async selectMockDevice() {
+    const elId = "device-item-" + this.deviceLike.id;
+    await waitForElementById(elId);
+    await tapById(elId);
+  }
+
+  async waitForSpinner() {
```

Comment:
i actually prefer the `async foo(){ await .. }` style because in that way, you don't need to explicit the typescript type (i don't want to have to write `: Promise<void>`